github_test.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"testing"
+	"os"
+)
+
+func TestGetListOfReleasesFromGitHubRepo(t *testing.T) {
+	cases := []struct {
+		repoUrl          string
+		lastReleaseTag   string
+		firstReleaseTag  string
+		gitHubOAuthToken string
+	}{
+		// These tests are obviously brittle. Suggestions welcome on a more durable way to test.
```

Comment:
One option is to use the golang GitHub API library, where someone else has written these tests :)